.github/workflows/test.yml

```diff
@@ -8,6 +8,12 @@ on:
       - '**'
       - '!doc/**'
 jobs:
+  vendor:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - run: make vendor && git diff --exit-code
```

Comment:
WDYT of doing a bunch of tidying here? `go mod tidy`, `go fmt ./...` and `go mod vendor` and then running `git diff`?

Reply:
So I updated the vendor target in the root Makefile in #590, [which prompted this PR](https://github.com/operator-framework/operator-registry/pull/590#discussion_r581070447) - but adding a linting checking the form of `go fmt ...` sounds like a step in the right direction. I don't have a strong opinion either way about whether the linting and vendor checks should be separate or their own combined test.

Reply:
In SDK, we group all of this under a [`make test-sanity`](https://github.com/operator-framework/operator-sdk/blob/34a55dd5ff27d9d87f96846eb5d17828d0e7b450/Makefile#L119-L126) target and then run that in [CI](https://github.com/operator-framework/operator-sdk/blob/34a55dd5ff27d9d87f96846eb5d17828d0e7b450/.github/workflows/test-sanity.yml#L34), which does what I'm describing. Basically all of the non-unit and non-e2e tests.

We found it nice having that in one make target so that its easy for developers to run it locally before pushing.

This is totally a nit though. I'm always good with incremental improvements.

Reply:
This kind of check may make more sense as a prow test rather than a github action too.

Reply:
Ah, gotcha yeah I'm fine with that rationale, although I typically prefer that something like `make test-sanity` just act as a driver and call individual make targets, like `make generate` and `make vendor` and verify the diff that way, but this is something that's pretty straight-forward to implement.

Reply:
I'm not too fussed with the details. The one bit of feedback we got from the community awhile back in SDK is that a sprawling Makefile made it hard to understand what contributors should run to verify their changes. _Maybe_ docs would help with that, but I feel like contributor docs have a way of getting out-of-date quickly.

On the other hand, it's nice having more focused make targets if you know you don't need to keep re-running tests irrelevant to your change.

I'm good either way.